frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1442,29 +1473,35 @@ function ReaderFooter:_updateFooterText(force_repaint, force_recompute)
         return
     end
     local text = self:genFooterText()
-    if text then
-        self.footer_text:setText(text)
-    end
     if self.settings.disable_progress_bar then
         if self.has_no_mode or not text or text == "" then
             self.text_width = 0
             self.footer_container.dimen.h = 0
             self.footer_text.height = 0
         else
+            self.footer_text:setMaxWidth(math.floor(self._saved_screen_width - 2 * self.settings.progress_margin_width))
+            self.footer_text:setText(text)
             self.text_width = self.footer_text:getSize().w
         end
         self.progress_bar.width = 0
     elseif self.settings.progress_bar_position then
         if text == "" then
+            self.footer_text:setText(text)
             self.footer_container.dimen.h = 0
             self.footer_text.height = 0
         end
+        self.footer_text:setMaxWidth(math.floor(self._saved_screen_width - 2 * self.settings.progress_margin_width))
+        self.footer_text:setText(text)
         self.progress_bar.width = math.floor(self._saved_screen_width - 2 * self.settings.progress_margin_width)
         self.text_width = self.footer_text:getSize().w
     else
         if self.has_no_mode or not text or text == "" then
             self.text_width = 0
+            self.footer_text:setText(text)
         else
+            local min_progress_bar_width = (100 - self.settings.progress_bar_min_width) / 100
```

Comment:
That local variable `min_progress_bar_width` is not what its name says :) it should be named something like `text_max_available_ratio`.